lib/iris/coords.py

```diff
@@ -1373,33 +1446,26 @@ def from_regular(cls, zeroth, step, count, standard_name=None,
             bounds values will be defined. Defaults to False.
 
         """
-        coord = DimCoord.__new__(cls)
-
-        coord.standard_name = standard_name
-        coord.long_name = long_name
-        coord.var_name = var_name
-        coord.units = units
-        coord.attributes = attributes
-        coord.coord_system = coord_system
-        coord.circular = circular
-
-        points = (zeroth+step) + step*np.arange(count, dtype=np.float32)
+        points = (zeroth + step) + step * np.arange(count, dtype=np.float32)
+        # Spoof `iris.util.is_regular`, which only works on coords (which we
+        # don't yet have).
+        diffs = np.diff(points)
+        if not np.allclose(diffs, np.mean(diffs), rtol=0.001):
+            points = (zeroth + step) + step * np.arange(count,
+                                                        dtype=np.float64)
```

Comment:
> The original state of this code was not a few assignments, it was a duplication of the functionality of `Coord.__init__`

The functionality of `Coord.__init__` _is_ just a few assignments, plus assignments to the points and bounds properties which can be avoided because we don't need the monotonitiy checks.

> When I changed the init this method no longer functioned

The reason this method no longer functioned was not because of changes to init (how could it be?) but because it was using the `_points` and `_bounds` attributes, which no longer exist.

> This version of this method is more robust because we are using the constructor for constructing the coord 

I agree with this, but this method is essentially a constructor by another name. You could have achieved the same robustness by factoring out a private method which does the duplicated functionality and which is called by both this method and the `__init__` method.